tests/queries/0_stateless/02669_memory_engine_exchange_tables.sql

```diff
@@ -0,0 +1,51 @@
+-- Tags: no-parallel
+
+DROP DATABASE IF EXISTS 02669_db1;
+DROP DATABASE IF EXISTS 02669_db2;
+CREATE DATABASE 02669_db1 ENGINE=Memory;
+CREATE DATABASE 02669_db2 ENGINE=Memory;
+CREATE TABLE 02669_db1.t1 (a UInt64) ENGINE=MergeTree ORDER BY a;
+CREATE TABLE 02669_db1.t2 (a UInt64) ENGINE=MergeTree ORDER BY a;
+
+-- Insert some data
+
+INSERT INTO 02669_db1.t1 VALUES (1);
+INSERT INTO 02669_db1.t2 VALUES (2);
+
+SELECT * FROM 02669_db1.t1;
+SELECT * FROM 02669_db1.t2;
+
+-- Usual RENAME
+
+RENAME TABLE 02669_db1.t1 TO 02669_db1.t3;
+SELECT * FROM 02669_db1.t3;
+RENAME TABLE 02669_db1.t3 TO 02669_db1.t1;
+
+-- EXCHANGE inside database
+
+EXCHANGE TABLES 02669_db1.t1 AND 02669_db1.t2;
+SELECT * FROM 02669_db1.t1;
+SELECT * FROM 02669_db1.t2;
+
+-- RENAME to another database
+
+RENAME TABLE 02669_db1.t1 TO 02669_db2.t1;
+SELECT * FROM 02669_db2.t1;
+
+-- EXCHANGE outside database
+
+EXCHANGE TABLES 02669_db2.t1 AND 02669_db1.t2;
+SELECT * FROM 02669_db2.t1;
+SELECT * FROM 02669_db1.t2;
+
+-- CREATE OR REPLACE
+
+CREATE OR REPLACE TABLE 02669_db1.t3 (a UInt64) ENGINE=MergeTree ORDER BY a;
+INSERT INTO 02669_db1.t3 VALUES (3);
+SELECT * FROM 02669_db1.t3;
+
+CREATE OR REPLACE TABLE 02669_db1.t3 (a UInt64) ENGINE=MergeTree ORDER BY a;
+SELECT * FROM 02669_db1.t3;
+
```

Comment:
Could you please add a test with a temporary table as well, and close https://github.com/ClickHouse/ClickHouse/pull/46050?